### PR TITLE
[DOC] Intercept request and response

### PR DESCRIPTION
**Pull request by gregurco**
| Q                | A
| ---------------- | -----
| Bug fix          | no
| New feature      | no
| BC breaks        | no
| Deprecations     | no
| Tests pass       | yes
| Fixed tickets    | -
| License          | MIT

See how it looks like: https://github.com/gregurco/EightPointsGuzzleBundle/blob/interceptor_doc/src/Resources/doc/intercept-request-and-response.md

### CODE REVIEW

File: src/Resources/doc/intercept-request-and-response.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        api_partners:
+            base_url: "http://pastners.tld"
+```
+
+and `api_partners` requires authorization using some token in header.  
+How we can do that?
+
+## Interceptor class
+
+First of all we have to crate out interceptor class, where will be all the logic:
```
Comment (florianpreusner) [inline src/Resources/doc/intercept-request-and-response.md]: create vs. create

File: src/Resources/doc/intercept-request-and-response.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+For example you are having configured next clients:
+
+```yaml
+eight_points_guzzle:
+    clients:
+        api_payment:
+            base_url: "http://api.domain.tld"
+        api_partners:
+            base_url: "http://pastners.tld"
```
Comment (florianpreusner) [inline src/Resources/doc/intercept-request-and-response.md]: pastners vs. partners

File: src/Resources/doc/intercept-request-and-response.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+     * @param string|null $serviceName
+     */
+    public function setServiceName($serviceName)
+    {
+        $this->serviceName = $serviceName;
+    }
+}
+```
+
+It's good to start from this initial file. We implemented `GuzzleEventListenerInterface` interface and added `setServiceName` method requested from interface.
```
Comment (florianpreusner) [inline src/Resources/doc/intercept-request-and-response.md]: Replace "It's good to start" by "Let's start", what do you think?

File: src/Resources/doc/intercept-request-and-response.md (hunk 2 above)
Comment (gregurco) [reply src/Resources/doc/intercept-request-and-response.md]: pastners comes from pasta :spaghetti: :smile:

**Pull request merged**